sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/AuthBundle.java

```diff
@@ -124,18 +127,48 @@ private Client createKeyLoaderClient(Environment environment, AuthConfig config,
   private KeySource createKeySources(KeyLocation keyLocation, Client client) {
     switch (keyLocation.getType()) {
       case PEM:
-        return new PemKeySource(keyLocation.getPemKeyId(), keyLocation.getLocation());
+        return new PemKeySource(
+            keyLocation.getPemKeyId(), keyLocation.getLocation(), keyLocation.getRequiredIssuer());
       case OPEN_ID_DISCOVERY:
+        validateKeyLocation(keyLocation.getLocation(), keyLocation.getRequiredIssuer());
         return new OpenIdProviderDiscoveryKeySource(
-            keyLocation.getLocation().toASCIIString(), client);
+            keyLocation.getLocation().toASCIIString(), client, keyLocation.getRequiredIssuer());
       case JWKS:
-        return new JwksKeySource(keyLocation.getLocation().toASCIIString(), client);
+        validateKeyLocation(keyLocation.getLocation(), keyLocation.getRequiredIssuer());
+        return new JwksKeySource(
+            keyLocation.getLocation().toASCIIString(), client, keyLocation.getRequiredIssuer());
       default:
         throw new IllegalArgumentException(
             "KeyLocation has no valid type: " + keyLocation.getType());
     }
   }
 
+  /**
+   * Validate, that if a required issuer is set as URI the host name of the key location source and
+   * the requiredIssuer must be the same.
+   *
+   * @param location The source {@link URI} of the key(s) as discovery or jwks endpoint.
+   * @param requiredIssuer The required issuer as {@link String} for the correlated key source.
+   */
+  private void validateKeyLocation(URI location, String requiredIssuer) {
+    if (StringUtils.isNotBlank(requiredIssuer) && StringUtils.contains(requiredIssuer, ':')) {
+      try {
+        URI issuerUri = new URI(requiredIssuer);
+        if (!StringUtils.equalsIgnoreCase(location.getHost(), issuerUri.getHost())) {
+          LOG.warn(
+              "The required issuer host name <{}> for the key <{}> does not match to the key"
+                  + " source uri host name <{}>.",
+              issuerUri.getHost(),
+              location,
+              location.getHost());
+        }
+      } catch (URISyntaxException e) {
+        throw new IllegalArgumentException(
+            "The requiredIssuer <" + requiredIssuer + "> is no valid stringOrURI", e);
```

Comment:
I would consider this case the same as not matching and log a warning. It may be a valid configuration because not every String with a colon is an URI.

Reply:
By specification it it :)

Reply:
👍  I'm fine with that, thanks for quoting the full spec.